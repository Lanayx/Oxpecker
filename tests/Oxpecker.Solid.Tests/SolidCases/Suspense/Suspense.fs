module Oxpecker.Solid.Tests.Cases.Suspense

open Oxpecker.Solid
open Browser

[<SolidComponent>]
let Test () =
    let posts, _ = createResource(fun () -> promise { return [| "Post1"; "Post2" |]})
    let trivia, _ = createResource(fun () -> promise { return [| "Trivia1"; "Trivia2" |]})

    SuspenseList(revealOrder="forwards", tail="collapsed"){
        Portal(mount=document.getElementById("postsHeader")) { h2() { "Posts"} }
        Suspense(fallback=h2() { "Loading posts..." }){
            ul() {
                For(each=posts.current){
                    yield fun post _ -> li() { post }
                }
            }
        }
        Suspense(fallback=h2() { "Loading trivia..." }){
            ul() {
                For(each=trivia.current){
                    yield fun trivia _ -> li() { trivia }
                }
            }
        }
    }
