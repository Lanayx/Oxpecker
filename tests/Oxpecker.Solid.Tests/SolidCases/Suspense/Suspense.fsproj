<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Fable.Promise" Version="3.2.0" />
  </ItemGroup>

  <ItemGroup>
    <Compile Include="Suspense.fs" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\..\..\..\src\Oxpecker.Solid\Oxpecker.Solid.fsproj" />
  </ItemGroup>

</Project>
