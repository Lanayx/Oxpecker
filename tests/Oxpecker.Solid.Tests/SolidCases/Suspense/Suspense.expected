import { createResource } from "solid-js";
import { PromiseBuilder__Delay_62FBFDE1, PromiseBuilder__Run_212F1D4B } from "./fable_modules/Fable.Promise.3.2.0/Promise.fs.js";
import { promise } from "./fable_modules/Fable.Promise.3.2.0/PromiseImpl.fs.js";

export function Test() {
    const patternInput = createResource(() => PromiseBuilder__Run_212F1D4B(promise, PromiseBuilder__Delay_62FBFDE1(promise, () => (Promise.resolve(["Post1", "Post2"])))), {});
    const patternInput_1 = createResource(() => PromiseBuilder__Run_212F1D4B(promise, PromiseBuilder__Delay_62FBFDE1(promise, () => (Promise.resolve(["Trivia1", "Trivia2"])))), {});
    return <SuspenseList revealOrder="forwards"
        tail="collapsed">
        <Portal mount={document.getElementById("postsHeader")}>
            <h2>
                Posts
            </h2>
        </Portal>
        <Suspense fallback={<h2>
                Loading posts...
            </h2>}>
            <ul>
                <For each={patternInput[0]()}>
                    {(post, _arg) => <li>
                        {post}
                    </li>}
                </For>
            </ul>
        </Suspense>
        <Suspense fallback={<h2>
                Loading trivia...
            </h2>}>
            <ul>
                <For each={patternInput_1[0]()}>
                    {(trivia_1, _arg_1) => <li>
                        {trivia_1}
                    </li>}
                </For>
            </ul>
        </Suspense>
    </SuspenseList>;
}

